### ISSUE TITLE
HS200-BL not discoverable

### ISSUE THREAD

**Issue by obviouspenguin** [plasticrake/homebridge-tplink-smarthome#346]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I purchased a new black colored version of the Kasa HS200. Connected it to WiFi and it’s showing on my Google home and Kasa apps. I cannot get it to appear in homebridge. I have a dozen other HS200 switches that work just fine. I’m not sure if this is a newer version of the HS200 switch as it initially paired with Bluetooth via the Kasa app. I’ve never had to do that with the others. 

### What, if anything, has recently changed?

Homebridge is up to date. Nothing has changed. The other switches are just fine. 

### Version

8.0.3

### Node Version

20.13.0

### Homebridge Version

1.8.1

### Which OS/Platform?

Linux

### Relevant log output

```shell
It only contains entries showing found cached devices. I searched though it and didn’t find anything about my new switch named Kitchen Ceiling Light
```

### Configuration

```json
{
            "name": "TplinkSmarthome",
            "addCustomCharacteristics": true,
            "switchModels": [
                "HS210",
                "HS200"
            ],
            "deviceTypes": [
                "plug"
            ],
            "devicesUseDiscoveryPort": false,
            "platform": "TplinkSmarthome"
        },
```

**Comment by mkarlsven** [comment]
I am having the exact same issue with mine. Only seems to affect my new HS 200 – BL devices.

**Comment by ericlytle** [comment]
Seeing the same issue. Have the HS 200 - BL

**Comment by jvinraviv** [comment]
Same here. Have HS200 with hardware version 5.26 (bought as a double switch HS200-2) while the other HS200 with hardware version 5.0 seem to work fine. Did TPLink change something in their protocol?

**Comment by obviouspenguin** [comment]
Any chance of getting this issue looked at please?

**Comment by miztahsparklez** [comment]
Seems like this also affects the WiFi only variants as well.  Got a new 3 pack white with hardware rev 5.26 on firmware 1.0.2.  One noticeable thing with the provisioning process is that it’s different and you also get some additional options that I haven’t seen before (auto turn off, led indicator mode change, delayed shut off, and default state).

wonder if they closed the hole to push people towards the matter/homekit switches?

**Comment by nordiclabs-io** [comment]
Also facing the exact same issue. My older HS200 switches are discovered just fine, the new HS200 BLE switches are not found. When I add them to manual discovery, the bridge just restarts itself every few seconds too.

**Comment by miztahsparklez** [comment]
Interesting note, I bought single packs instead with hardware rev 5.6 and they behave like normal.

**Comment by picnicbob** [comment]
Same issue here with manual IP configured, although unlike @nordiclabs-io my bridge doesn't restart.

**Comment by miztahsparklez** [comment]
I suspect these have slightly different hardware inside.  I was able to work around using Tapo plugin, but I wonder if these need to be added to the allow list on the plugin or if the port on these needed to be open to allow local access.

**Comment by devicemodder** [comment]
@miztahsparklez do you have a tutorial for that?

**Comment by VickY0E** [comment]
Same here with HS200 wifi, HW version 5.26, Firmware 1.0.2, even manual discovery doesn't work while all the other switches are working just fine on auto discovery

**Comment by VickY0E** [comment]
Did port scan on HS200 wifi, HW version 5.26, Firmware 1.0.2, port 80 is open instead of the usual 9999, when I set manual discovery to port 80, the child bridge just keeps restarting, I think TP-link changed the protocol or something, this is terrible at the moment

**Comment by miztahsparklez** [comment]
> @miztahsparklez do you have a tutorial for that?

Not really.  Download Tapo plugin in HB, setup credentials, use Tapo app to setup switch, add IP addresss to Tapo plugin.

There was another article I found that was exploring opening up the correct port by SSH to the switch, but I never got that far, as I already fixed my problem by changing out my problem switches to the newer single packs.  The “desirable” 3 pack is the brown box variant.  If you have the color graphic 3 pack, you likely have the troublesome pack.  I haven’t seen a 1 pack do this yet, but I suspect if it’s made from the same batch, you might see the same problem.

**Comment by ctrlaltdel17** [comment]
Any updates? I picked up H/W version 5.26 switches and they aren't being detected by the plugin either.

**Comment by xhi-nico** [comment]
seems like this plugin needs to be updated to ensure the new kasa devices are compatible. Is this plugin still active?

**Comment by jlayman81** [comment]
Same issue here. 5.26, 1.0.2. My whole house is running Kasa switches and only these new ones have the issue.

**Comment by jaredhimself** [comment]
Just installed a new 3-pack. Having the same issue. Has anyone found a resolution with the Kasa plugin, yet?

**Comment by jlayman81** [comment]
I have a bag of these switches to finish out my house sitting in my garage because of this issue. Have over 20 already installed that work great. It’s just the new ones that aren’t seen by homebridge. With all the effort and DIY home automation, roadblocks like this just take the wind right out. How many houses can’t finish their Kasa setup because of this? 

This issue was raised in May and is still unresolved. It looks like support for this plugin is dead. Gotta hope for a new one or create a big pile of e-waste and pony up for switches with native HomeKit integration.

If that happens, RIP homebridge.

**Comment by xhi-nico** [comment]
I made the mistake to install newer models of switches across the house; was deeply disappointed when I found out this plugin is not updated for newer models. 

I switched to HomeAssistant JUST for these devices, the Kasa plugin there is up-to-date and is working with the newer devices. I have also realized how much better/simpler Homebridge compared to HA. 

Here's hoping this plugin is updated so I can switch back.

**Comment by DustonF** [comment]
I'm also in the same boat, just bought a new switch. Unable to discovery auto or manual. Please and thank you if someone could look into this. Hardware version on the switches is 5.26, Firmware 1.0.3.

**Comment by Dpd136** [comment]
Anyone figure out a fix? I’m new to homebridge, this was driving me crazy

**Comment by jlayman81** [comment]
There are a couple of workarounds in this thread. One is switching to home assistant, the other is using tapo. Both are described by other posters here. 

As far as a fix is concerned, I haven’t seen the developer make any comments that they are working on it. Sad times.

**Comment by picnicbob** [comment]
For those questioning which Tapo plugin to try I couldn't get things to work with the verified "homebridge-tp-link-tapo" but I did get it to work with "homebridge-tapo." I linked my Kasa account in the Tapo app and used manual device discovery (by IP address) in Tapo plugin settings. My old style Kasa switches are still in homebridge-tplink-smarthome and the new hardware I have is in homebridge-tapo. So far so good.

**Comment by fredsocial** [comment]
Same here

**Comment by dfizzo** [comment]
I am dealing with same issue and tried homebridge-tapo work around. But, on homebridge-tapo, while I am able to discover 5.26 switches, there is no response for either in HomeKit. Anyone experience a similar issue?

**Comment by Texnick19** [comment]
> For those questioning which Tapo plugin to try I couldn't get things to work with the verified "homebridge-tp-link-tapo" but I did get it to work with "homebridge-tapo." I linked my Kasa account in the Tapo app and used manual device discovery (by IP address) in Tapo plugin settings. My old style Kasa switches are still in homebridge-tplink-smarthome and the new hardware I have is in homebridge-tapo. So far so good.

How did you configure it for Kasa?  

 "name": "Tapo Smart Platform",
    "username": "xxxxxxx",
    "password": "xxxxxxxx",
    "devices": [
        {
            "name": "Kitchen Main",
            "host": "192.168.1.157",
            "updateInterval": 10,
            "timeout": 2
        },
        {
            "type": "plug",
            "updateInterval": 10,
            "timeout": 2
        }
    ],
    "platform": "TapoP100",
    "_bridge": {
        "username": "0E:4B:59:1A:4A:41",
        "port": 39489
    }
}

**Comment by martinlaoxu** [comment]
Tapo plugin wroks with it. Just need to setup the ip of the switch as below. 

<img width="775" height="561" alt="Image" src="https://github.com/user-attachments/assets/0fef08b7-e9e3-4edc-b8aa-cc624041df4c" />

**Comment by martinlaoxu** [comment]
> This drove me insane. I FINALLY found a workaround. Not elegant - but I only have one hs200 switch
> 
> Add HS200 to kasa app
> Create Alexa account and link to Kasa app - should see all your switches
> Homekit: Install Homebridge Alexa Smart Home @joeyhage. use homebridge.local and port is anything you want
> Amazon: **this was holding me up and was critical for me** turn on MFA with an authenticator app (NOT sms/email)
> Homekit: open the plugin log and follow directions to authenticate. Use Firefox or chrome in incognito (safari issues)
> 
> this worked for me 
> 

Use Tapo app (shares the same account with Kasa), then the Tapo plugin works with it.

**Comment by jbc1989** [comment]
I could not get it to work with the tapo app and the newest firmware. I
would add my credentials, add static IP for the switch, and it would never
find it. Not sure if I was missing something
